src/Lab.java

```diff
@@ -1,10 +1,17 @@
+import java.io.BufferedReader;
+import java.io.FileReader;
 import java.io.IOException;
 import java.net.*;
 import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
 
 
 public class Lab {
-    public static void main(String[] args) throws IOException{
+    public static void main(String[] args) throws IOException {
+        Set<String> blockList = new HashSet<>();
+        if (args.length > 0)
+            blockList = launchBlockList(args[0]);
```

Comment:
a text file with blocked domains should be loaded before receiving the query from user